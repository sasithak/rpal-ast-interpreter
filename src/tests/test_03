let 
.and 
..rec 
...function_form 
....<ID:int_len> 
....<ID:n> 
....-> 
.....eq 
......<ID:n> 
......<INT:0> 
.....<INT:0>
.....+
......gamma
.......<ID:int_len>
......./
........<ID:n>
........<INT:10>
......<INT:1> 
..rec
...function_form
....<ID:int_reverse>
....,
.....<ID:p>
.....<ID:size>
....->
.....eq
......<ID:p>
......<INT:0>
.....<INT:0>
.....+
......gamma
.......<ID:int_reverse>
.......tau
......../
.........<ID:p>
.........<INT:10>
........-
.........<ID:size>
.........<INT:1>
......* 
.......-
........<ID:p>
........*
........./
..........<ID:p>
..........<INT:10>
.........<INT:10>
.......**
........<INT:10>
........<ID:size>
.let
..rec
...function_form
....<ID:is_palindrome_int> 
....<ID:p>
....->
.....ls
......<ID:p>
......<INT:0>
.....<false>
.....eq
......<ID:p>
......gamma
.......<ID:int_reverse>
.......tau
........<ID:p>
........-
.........gamma
..........<ID:int_len>
..........<ID:p>
.........<INT:1>
..let
...rec
....function_form
.....<ID:palindrome_range>
.....,
......<ID:a>
......<ID:b>
.....->
......gr
.......<ID:a>
.......<ID:b>
......<nil>
......->
.......gamma
........<ID:is_palindrome_int>
........<ID:b>
.......aug
........gamma
.........<ID:palindrome_range>
.........tau
..........<ID:a>
..........-
...........<ID:b>
...........<INT:1>
........<ID:b>
.......gamma
........<ID:palindrome_range>
........tau 
.........<ID:a>
.........-
..........<ID:b>
..........<INT:1>
...gamma
....<ID:Print>
....gamma
.....<ID:palindrome_range>
.....tau
......<INT:10>
......<INT:200>
