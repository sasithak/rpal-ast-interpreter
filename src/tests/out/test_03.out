(11, 22, 33, 44, 55, 66, 77, 88, 99, 101, 111, 121, 131, 141, 151, 161, 171, 181, 191)
